app/models/source.rb

```diff
@@ -17,22 +17,30 @@ class Source < ActiveRecord::Base
 
   has_many :images, through: :attachments,
                     source: :asset,
-                    source_type: 'Image'
+                    source_type: 'Image',
+                    after_add: :touch_self,
+                    before_remove: :touch_self
 
   has_many :large_images, -> { where size: 'large' },
                           through: :attachments,
                           source: :asset,
-                          source_type: 'Image'
+                          source_type: 'Image',
+                          after_add: :touch_self,
+                          before_remove: :touch_self
 
   has_many :small_images, -> { where size: 'small' },
                           through: :attachments,
                           source: :asset,
-                          source_type: 'Image'
+                          source_type: 'Image',
+                          after_add: :touch_self,
+                          before_remove: :touch_self
 
   has_many :thumbnails, -> { where size: 'thumbnail' },
                         through: :attachments,
                         source: :asset,
-                        source_type: 'Image'
+                        source_type: 'Image',
+                        after_add: :touch_self,
+                        before_remove: :touch_self
```

Comment:
It's not ideal here to be re-asserting the `after_add` and `before_remove` statements for `images`, `large_images`, `small_images`, and `thumbnails`.  Really, large_image, small_image, and thumbnail should be refactored to be model methods instead of relations, for example:

     def thumbnail
        images.where(size: 'thumbnail')
      end

That way the `after_add` and `before_remove` would only need to be asserted for `images`.  But that is out of scope for this PR.